lib/rouge/lexers/mason.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Mason < TemplateLexer
+      title 'Mason'
+      desc 'HTML::Mason'
+      tag 'mason'
+      filenames '*.mi', '*.mc', '*.mas', '*.m', '*.mhtml', '*.mcomp', 'autohandler', 'dhandler'
+      mimetypes 'text/x-mason', 'application/x-mason'
+
+      def initialize(*)
+        super
+        @perl = Perl.new
+      end
+  
+      def self.detect?(text)
+        return false if text.doctype?(/[ht|x]ml/)
+        return true if text.doctype?
+      end
+
+      # Note: If you add a tag in the lines below, you also need to modify "disambiguate '*.m'" in file disambiguation.rb
+      textblocks = %w(text doc)
+      perlblocks = %w(args flags attr init once shared perl cleanup filter)
+      components = %w(def method)

```

Comment:
These declarations will perform better if they're constants. 
```ruby
      TEXT_BLOCKS = %w(text doc)
      PERL_BLOCKS = %w(args flags attr init once shared perl cleanup filter)
      COMPONENTS  = %w(def method)
```
lowercase named variables are deemed as "local" by the Garbage Collector and will be cleaned away once after first pass. Constants are retained for the lifetime.

Reply:
Nice to know! Applied.

Reply:
@pyrmont Inspite of your aversion to using constants, I recommended them anyways in order to avoid confusion regarding the scope of a `self.textblocks` method. Apologies.

Reply:
@ashmaroli No, that's fine. My concern about using constants would rather be the fact that they're _not_ garbage collected and so will sit around in memory but I'm not sure how significant this issue is given that Rouge is not designed to be a long-running process. For the purposes of this PR, I think it's fine.

Reply:
> given that Rouge is not designed to be a long-running process

When a library like Jekyll uses Rouge, there may be numerous documents that contain code-blocks of a certain language. So in theory, GC should not be sweeping away these arrays only to be re-allocated on the next encounter.

Of course, all of the above is just my notion and have no concrete proof to say that's how Ruby works..